File: parser/src/main/java/dev/mvmo/sicklang/parser/ast/node/expression/IdentifierExpressionNode.java
package dev.mvmo.sicklang.parser.ast.node.expression;

import dev.mvmo.sicklang.token.Token;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor(staticName = "newInstance")
public class IdentifierExpressionNode implements ExpressionNode {

    private final Token token;
    private final String value;

    @Override
    public String tokenLiteral() {
        return token.literal();
    }

}

File: parser/src/main/java/dev/mvmo/sicklang/parser/ast/expression/IdentifierExpressionNode.kt
package dev.mvmo.sicklang.parser.ast.expression

import dev.mvmo.sicklang.token.Token

class IdentifierExpressionNode(val token: Token, val value: String) : ExpressionNode {

    override fun tokenLiteral(): String =
        token.literal()

    override fun toString(): String =
        value

}